### PR TITLE
Build mscorlib with CoreCLR MSBuild on Linux and Mac

### PR DESCRIPTION
**Pull request by JohnChen0**
Address issue #1934 by using CoreCLR MSBuild to build mscorlib.dll on Linux and Mac, and calling crossgen to generate mscorlib.ni.dll.

### CODE REVIEW

File: src/mscorlib/Tools/PostProcessingTools.targets [hunk 1]
```diff
@@ -21,7 +21,7 @@
     
     <!-- Copy to the final output location -->
     <Copy Retries="3" SourceFiles="@(RewrittenAssembly)" DestinationFiles="$(FinalOutputPath)\%(RewrittenAssembly.FileName)%(RewrittenAssembly.Extension)"/>
-    <Message Importance="High" Text="$(MSBuildProjectName) -&gt; $(FinalOutputPath)\%(RewrittenAssembly.FileName)%(RewrittenAssembly.Extension)" />
+    <Message Importance="High" Text="$(MSBuildProjectName) -&gt; $(FinalOutputPath)%(RewrittenAssembly.FileName)%(RewrittenAssembly.Extension)" />
```
Comment (ellismg) [inline src/mscorlib/Tools/PostProcessingTools.targets]: This always bugged me.  Thanks for fixing.

**Comment by ellismg** [comment]
The CI isn't going to give you coverage that this actually works (since the legs pass "skipmscorlib").  Once this lands, we'll need need to enable it in CI for Ubuntu 14.04 and OSX (and then figure out a plan to bring the other platforms online, we already have CLI packages for CentOS so that should be "easy").

**Comment by ellismg** [comment]
We'll also need to update our VSO legs to turn this on there (@chcosta or I can help you with that).

**Comment by JohnChen0** [comment]
I have verified that build works after removing `/p:ToolNugetRuntimeId=$_ToolNugetRuntimeId`. I will make that change.

**Comment by JohnChen0** [comment]
@ellismg: Thanks for the feedback. I have updated the PR to address the issues.

**Comment by ellismg** [comment]
Thanks, LGTM!

**Comment by JohnChen0** [comment]
@ellismg and @chcosta: CoreCLR repo should now have all the code needed to create mscorlib.dll and mscorlib.ni.dll on Linux and Mac. Could you help with updating CI and VSO? Thanks.

**Comment by chcosta** [comment]
Ack, I'll follow up with the VSO aspect.

**Pull request merged**